bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastBridgeHandler.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.yamahamusiccast.internal;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.yamahamusiccast.internal.model.UdpMessage;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.thing.*;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link YamahaMusiccastBridgeHandler} is responsible for dispatching UDP events to linked Things.
+ *
+ * @author Lennert Coopman - Initial contribution
+ */
+@NonNullByDefault
+public class YamahaMusiccastBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(YamahaMusiccastBridgeHandler.class);
+
+    private final ScheduledExecutorService udpScheduler = ThreadPoolManager
+            .getScheduledPool("YamahaMusiccastListener" + "-" + thing.getUID().getId());
+    private @Nullable ScheduledFuture<?> listenerJob;
+    private final UdpListener udpListener;
+
+    public YamahaMusiccastBridgeHandler(Bridge bridge) {
+        super(bridge);
+        udpListener = new UdpListener(this);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.ONLINE);
+        startUDPListenerJob();
+    }
+
+    @Override
+    public void dispose() {
+        stopUDPListenerJob();
+        super.dispose();
+    }
+
+    private void startUDPListenerJob() {
+        logger.info("YXC - Bridge Listener to start in 5 seconds");
+        listenerJob = udpScheduler.schedule(udpListener, 5, TimeUnit.SECONDS);
+    }
+
+    private void stopUDPListenerJob() {
+        if (listenerJob != null) {
+            listenerJob.cancel(true);
+            udpListener.shutdown();
+            logger.debug("Canceling listener job");
+        }
+    }
+
+    public void handleUDPEvent(String json) {
+        String udpDeviceId = "";
+        Bridge bridge = (Bridge) thing;
+        for (Thing thing : bridge.getThings()) {
+            ThingStatusInfo statusInfo = thing.getStatusInfo();
+            switch (statusInfo.getStatus()) {
+                case ONLINE:
+                    logger.debug("Thing Status: ONLINE - {}", thing.getLabel());
+
+                    YamahaMusiccastHandler handler = (YamahaMusiccastHandler) thing.getHandler();
+                    logger.debug("UDP: {} - {} ({})", json, handler.getDeviceId(), thing.getLabel());
+                    try {
+                        @Nullable
+                        UdpMessage targetObject = new Gson().fromJson(json, UdpMessage.class);
```

Comment:
You could make the Gson object a field to save resources.

Reply:
Do you have an example of that? Everything I have found boils down to the same so I must be missing something. Thx.

Reply:
The Gson object is created newly every time the execution path comes here. You can assign `new Gson()` to a class field, so that it is only created once, when the class is instantiated.

Reply:
Thx, should have found that myself.
Prepared changes, will be available with next push